experimental/docs/gdrive_backend.md

```diff
@@ -0,0 +1,229 @@
+# Google Drive Backend for Ragas
+
+The Google Drive backend allows you to store Ragas datasets and experiments in Google Sheets within your Google Drive. This provides a cloud-based, collaborative storage solution that's familiar to many users.
+
+## Features
+
+- **Cloud Storage**: Store your datasets and experiments in Google Drive
+- **Collaborative**: Share and collaborate on datasets using Google Drive's sharing features
+- **Google Sheets Format**: Data is stored in Google Sheets for easy viewing and editing
+- **Automatic Structure**: Creates organized folder structure (datasets/ and experiments/)
+- **Type Preservation**: Attempts to preserve basic data types (strings, numbers)
+- **Multiple Authentication**: Supports both OAuth and Service Account authentication
+
+## Installation
+
+```bash
+# Install with Google Drive dependencies
+pip install "ragas_experimental[gdrive]"
+```
+
+## Setup
+
+### 1. Google Cloud Project Setup
+
+1. Go to the [Google Cloud Console](https://console.cloud.google.com/)
+2. Create a new project or select an existing one
+3. Enable the following APIs:
+   - Google Drive API
+   - Google Sheets API
+
+### 2. Authentication Setup
+
+Choose one of two authentication methods:
+
+#### Option A: OAuth 2.0 (Recommended for Development)
+
+1. In Google Cloud Console, go to "Credentials"
+2. Click "Create Credentials" → "OAuth client ID"
+3. Choose "Desktop application"
+4. Download the JSON file (save as `credentials.json`)
+
+#### Option B: Service Account (Recommended for Production)
+
+1. In Google Cloud Console, go to "Credentials"
+2. Click "Create Credentials" → "Service account"
+3. Create the service account and download the JSON key file
+4. Share your Google Drive folder with the service account email
```

Comment:
was following this and I'm not sure if OAuth is needed

Reply:
we can keep it there but I think we should use the service account as the recommended approach because it works well for scripts right